File: WOT-iOS/WOT-iOS/core/datasources/WOTTree/WOTTree+Pivot/WOTPivotRowNode.m
//
//  WOTPivotRowNode.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 8/27/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTPivotRowNode.h"
#import "WOTNode+Enumeration.h"

@implementation WOTPivotRowNode

- (PivotStickyType)stickyType {
    return PivotStickyTypeHorizontal;
}

@end

@implementation WOTPivotRowNode (Dimension)

- (NSInteger)y {

    NSInteger result = 0;
    result += [WOTNodeEnumerator.sharedInstance childrenWidthWithSiblingNode:self orValue:1];
    result += self.dimensionDelegate.rootNodeHeight;
    return result;
}

- (NSInteger)x {
    return [WOTNodeEnumerator.sharedInstance visibleParentsCountWithNode:self];
}

- (NSInteger)width {

    return 1;
}

- (NSInteger)height {
    NSArray *endpoints = [WOTNodeEnumerator.sharedInstance endpointsWithNode: self];
    return endpoints.count;

}

@end

File: WOT-iOS/WOT-iOS/core/datasources/WOTTree/WOTDataModel/WOTPivotDataModel/WOTPivotNodes/extensions/WOTPivotRowNode.swift
//
//  WOTPivotRowNode.swift
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 7/17/18.
//  Copyright © 2018 Pavel Yeshchyk. All rights reserved.
//

import Foundation

class WOTPivotRowNode: WOTPivotNode {

    override var stickyType: PivotStickyType {

        return PivotStickyType.horizontal
    }

    override var cellType: WOTPivotCellType {
        return .row
    }
}
